drop-in/src/main/java/com/adyen/checkout/dropin/internal/ui/CardComponentDialogFragment.kt

```diff
@@ -65,6 +59,20 @@ internal class CardComponentDialogFragment : BaseComponentDialogFragment(), Addr
             .launchIn(viewLifecycleOwner.lifecycleScope)
     }
 
+    private fun initToolbar() = with(binding.bottomSheetToolbar) {
```

Comment:
I found an edge case scenario that displays the wrong icon:
- Load drop-in with only one payment method that can display multiple screens (card + address lookup or BACS)
- Configure drop-in with `setSkipListWhenSinglePaymentMethod(true)` 
- Open drop-in and open the second screen
- A back icon is expected to show but a close icon is actually shown
- Tapping the icon does go back to the first screen so the tap action works correctly

I don't think we can do this now without adding extra functionality, let's discuss whether it's worth it to handle this

Reply:
Good point. I did not like leaving it like that, so I fixed it. Could you please test it again?

Reply:
Great effort with this! I like the general idea but my main issue with the implementation is that it only works for us internally (drop-in). Merchants using standalone components won't be able to benefit from it (as we're observing `component.viewFlow` which is internal). It would be great if this became part of some public functionality. 

Personally I wouldn't mind if we keep this part out of the scope of this PR as it's getting quite big and all of the solutions are going to be complicated enough to deserve their own PR.

Reply:
After discussing with design and product, we decided to leave it as it is for now.